### ISSUE TITLE
sudo install

### ISSUE THREAD

**Issue by casperdcl** [CCPPETMR/SIRF-SuperBuild#121]
from @rijobro https://github.com/CCPPETMR/SIRF/issues/179

We should be able to:

```bash
cmake -DCMAKE_INSTALL_PREFIX=/usr/local .
make -j preinstall  # note that this actually does install. seems counterintuitve. there's no `build` target
sudo make install
```

this currently throws errors due to `root` permissions required for `preinstall`/build phases.

- [ ] change `configure_file()` to intermediate directory and `install()` from there
- [ ] fix other issues (e.g. `ismrmrd`)
- [ ] create a `build` target?

**Comment by KrisThielemans** [comment]
I think this is hopeless for a Superbuild project. All projects are built/installed directly. Changing this would need major surgery in the CMake `ExternalProject` stuff.

We should just close this, and just add a note in the INSTALL doc.

**Comment by rijobro** [comment]
This doesn't seem to be the same problem as CCPPETMR/SIRF#179 anyway. On the SIRF side, the problem was that files were being copied during the `configure` step, as opposed to the `make` step here.

**Comment by KrisThielemans** [comment]
ok, @paskino or @rijobro, could you just update the INSTALL to say "either change the prefix, or use `sudo make`" (but say it better!) and let that commit close this issue

**Comment by casperdcl** [comment]
We can fix (have already I think?) configuring into an intermediate directory, the problem as far as I can remember is some external dependencies don't.

**Comment by KrisThielemans** [comment]
we've fixed SIRF. The problem here is the SuperBuild. just `make` will actually imply `install` for `ExternalProject`. Let's not go there...

**Comment by rijobro** [comment]
We're getting confused here. We fixed the SIRF problem by splitting the `configure` and `install` steps. In the superbuild, we don't install to `/usr/local` by default (instead `${CMAKE_BINARY_DIR}/INSTALL`, so that won't require `sudo` either.

I think we can simply close this issue.

**Comment by KrisThielemans** [comment]
It just needs a line in the installation documentation, as I wrote above.

**Comment by rijobro** [comment]
closed via cee0027275df20ae32f490c84c0e7c79864fe0da.
